### ISSUE TITLE
[a11y] [Choice list] Error message not programmatically clear

### ISSUE THREAD

**Issue by dpersing** [Shopify/polaris-react#1589]
## Issue summary

The choice list `<fieldset>` output includes an `aria-describedby` value that points to the `id` of an element that is used to display error messages for the choice list. This is currently implemented with a few issues:

- The `aria-describedby` attribute points to a missing `id` by default when the error message is not displayed, making it invalid
- `aria-describedby` is not conveyed consistently unless it's on a focusable element, and screen reader users may miss it since it's not tied to the fields themselves ([see an overview of `aria-describedby` and related attribute behaviors from September 2018](https://www.davidmacd.com/blog/does-aria-label-override-static-text.html))

## Expected behavior

- `aria-describedby` and similar attributes will not point to invalid or missing `id` values
- Inline error messages will be tied to individual fields in error

## Actual behavior

- The fieldset's `aria-describedby` points to a missing `id` by default
- The `aria-describedby` is not tied to a focusable element

```
<fieldset class="Polaris-ChoiceList" id="ChoiceList1" aria-invalid="false" aria-describedby="ChoiceList1Error">
<legend class="Polaris-ChoiceList__Title">Company name</legend>
...
</fieldset>
```

## Steps to reproduce the problem

1. Visit any example of the choice list: https://polaris.shopify.com/components/forms/choice-list
1. Use the web inspector to view the output of the example `<fieldset>` elements

## Recommendation

- Remove the `aria-describedby` from the `<fieldset>`
- When errors are displayed, add `aria-describedby` to any relevant fields so the error can be read on focus for each field

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 3.16.0
- Browser: All
- Device: All
- Operating System: All
